dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/PeriodTypeController.java

```diff
@@ -32,68 +32,52 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import lombok.RequiredArgsConstructor;
+
 import org.hisp.dhis.common.DhisApiVersion;
-import org.hisp.dhis.fieldfilter.FieldFilterParams;
-import org.hisp.dhis.fieldfilter.FieldFilterService;
-import org.hisp.dhis.node.NodeUtils;
-import org.hisp.dhis.node.Preset;
-import org.hisp.dhis.node.types.RootNode;
+import org.hisp.dhis.commons.jackson.domain.JsonRoot;
+import org.hisp.dhis.fieldfiltering.FieldFilterManager;
+import org.hisp.dhis.fieldfiltering.FieldFilterParams;
 import org.hisp.dhis.period.PeriodService;
 import org.hisp.dhis.period.RelativePeriodEnum;
 import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
-import org.hisp.dhis.webapi.service.ContextService;
 import org.hisp.dhis.webapi.webdomain.PeriodType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.google.common.collect.Lists;
+import com.fasterxml.jackson.databind.node.ObjectNode;
 
 /**
  * @author Morten Olav Hansen <mortenoh@gmail.com>
  */
 @RestController
 @ApiVersion( { DhisApiVersion.DEFAULT, DhisApiVersion.ALL } )
 @RequestMapping( value = "/periodTypes" )
+@RequiredArgsConstructor
 public class PeriodTypeController
 {
     private final PeriodService periodService;
 
-    private final ContextService contextService;
-
-    private final FieldFilterService fieldFilterService;
-
-    public PeriodTypeController( PeriodService periodService, ContextService contextService,
-        FieldFilterService fieldFilterService )
-    {
-        this.periodService = periodService;
-        this.contextService = contextService;
-        this.fieldFilterService = fieldFilterService;
-    }
+    private final FieldFilterManager fieldFilterManager;
```

Comment:
Seeing this example I think `Service` suffix is almost better than `Manager`. I think you also switched to `@Service` annotation.